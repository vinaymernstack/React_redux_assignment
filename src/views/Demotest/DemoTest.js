import React, { useRef, useState } from 'react';

const getSum = (value1, value2) => {
  return value1 + value2;
};

const Demotest = () => {
  const inputValueOne = useRef(null);
  const inputValueTwo = useRef(null);
  const [sum, setSum] = useState(null);

  const add = () => {
    const result = getSum(
      parseFloat(inputValueOne.current.value),
      parseFloat(inputValueTwo.current.value),
    );

    setSum(result);
  };

  return (
    <form>
      <input type="number" defaultValue={0} ref={inputValueOne} />
      <input type="number" defaultValue={0} ref={inputValueTwo} />
      <button type="button" onClick={add} aria-label="Add" />
      <p className="result">{sum}</p>
    </form>
  );
};

export { getSum };
export default Demotest;