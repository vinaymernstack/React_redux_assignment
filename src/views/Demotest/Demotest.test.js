import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import Demotest, { getSum } from './DemoTest.js';

describe('Calculator component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(<Demotest />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return the correct sum', () => {
    const sum = getSum(3, 5);
    expect(sum).toEqual(8);
  });

  it('should render required form elements', () => {
    const calculator = shallow(<Demotest />);

    const form = calculator.find('form');
    expect(form.length).toBe(1);
    expect(form.find('input').length).toBe(2);
    expect(form.find('button').length).toBe(1);
    expect(form.find('p.result').length).toBe(1);
  });

  it('should display the result on add', () => {
    const calculator = mount(<Demotest />);

    const form = calculator.find('form');

    form.childAt(0).instance().value = 3;
    form.childAt(1).instance().value = 5;
    form.find('button').simulate('click');

    const result = calculator.find('.result');
    expect(result.text()).toEqual('8');
  });
});