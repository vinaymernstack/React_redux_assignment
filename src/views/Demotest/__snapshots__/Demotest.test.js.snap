// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should render snapshot 1`] = `
<form>
  <input
    defaultValue={0}
    type="number"
  />
  <input
    defaultValue={0}
    type="number"
  />
  <button
    aria-label="Add"
    onClick={[Function]}
    type="button"
  />
  <p
    className="result"
  />
</form>
`;
